File: backend/src/models/Ride.js
// src/models/Ride.js - Ride schema
const mongoose = require('mongoose');

const rideSchema = new mongoose.Schema({
  rideId: {
    type: String,
    unique: true,
    required: true
  },
  riderId: {
    type: String,
    required: true
  },
  driverId: String,
  vehicleId: String,
  pickupLat: {
    type: Number,
    required: true
  },
  pickupLng: {
    type: Number,
    required: true
  },
  dropoffLat: {
    type: Number,
    required: true
  },
  dropoffLng: {
    type: Number,
    required: true
  },
  pickupAddress: {
    type: String,
    required: true
  },
  dropoffAddress: {
    type: String,
    required: true
  },
  vehicleType: {
    type: String,
    enum: ['ECONOMY', 'PREMIUM', 'SUV', 'AUTO', 'BIKE'],
    required: true
  },
  estimatedFare: {
    type: Number,
    required: true
  },
  actualFare: {
    type: Number,
    default: 0
  },
  status: {
    type: String,
    enum: ['PENDING', 'ACCEPTED', 'STARTED', 'COMPLETED', 'CANCELLED'],
    default: 'PENDING'
  },
  rideStartTime: Date,
  rideEndTime: Date,
  rating: Number,
  feedback: String,
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Ride', rideSchema);

File: android-app/app/src/main/java/com/minovox/goavehiclebooking/models/Ride.kt
// Ride.kt - Ride model
package com.minovox.goavehiclebooking.models

data class Ride(
    val rideId: String,
    val riderId: String,
    val driverId: String? = null,
    val vehicleId: String? = null,
    val pickupLat: Double,
    val pickupLng: Double,
    val dropoffLat: Double,
    val dropoffLng: Double,
    val pickupAddress: String,
    val dropoffAddress: String,
    val vehicleType: String, // ECONOMY, PREMIUM, SUV, AUTO, BIKE
    val estimatedFare: Double,
    val actualFare: Double = 0.0,
    val status: String, // PENDING, ACCEPTED, STARTED, COMPLETED, CANCELLED
    val rideStartTime: Long? = null,
    val rideEndTime: Long? = null,
    val rating: Int? = null,
    val feedback: String? = null,
    val createdAt: Long
)